v3/position.go

```diff
@@ -1,14 +1,19 @@
 package geohex
 
-import "math"
+import (
+	"fmt"
+	"math"
+)
 
 // Position implements a grid tile position
 type Position struct {
 	X, Y int
 	z    *Zoom
+
+	code string
```

Comment:
I'm sorry, I really don't like this. This is a base library and we should not add any hidden memoization to it. I understand that you want to cache the result, but it's quite specific to your implementation and you should add this to your own structs, something like:

```
package mypkg

type Position struct {
  geohex.Position
  code string
}

func (p *Position) Code() string {
  if p.code == "" { 
    p.code = p.Position.Code()
  }
  return p.code
}
```